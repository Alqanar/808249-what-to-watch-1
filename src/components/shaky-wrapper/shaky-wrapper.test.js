import React from "react";
import renderer from "react-test-renderer";

import ShakyWrapper from "./shaky-wrapper.tsx";


describe(`wrapper correctly renders`, () => {
  it(`when needVanish passed renders without wrapper`, () => {
    const tree = renderer
      .create(
        <ShakyWrapper needVanish>
          <p>Hello, world!</p>
        </ShakyWrapper>

      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`when needVanish not passed renders div wrapper`, () => {
    const tree = renderer
      .create(
        <ShakyWrapper>
          <p>Hello, world!</p>
        </ShakyWrapper>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
