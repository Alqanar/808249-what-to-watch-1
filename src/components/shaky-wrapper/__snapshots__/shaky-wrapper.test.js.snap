// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wrapper correctly renders when needVanish not passed renders div wrapper 1`] = `
<div>
  <p>
    Hello, world!
  </p>
</div>
`;

exports[`wrapper correctly renders when needVanish passed renders without wrapper 1`] = `
<p>
  Hello, world!
</p>
`;
